lib/hal/video.c

```diff
@@ -341,18 +336,28 @@ void XVideoInit(DWORD dwMode, int width, int height, int bpp)
 
 	XVideoSetVideoEnable(FALSE);
 
+	PVOID framebufferMemory = AvGetSavedDataAddress();
+
 	if (framebufferMemory != NULL) {
-		MmPersistContiguousMemory(framebufferMemory, screenSize, FALSE);
-		MmFreeContiguousMemory(framebufferMemory);
+		SIZE_T frameBufferSize = MmQueryAllocationSize(framebufferMemory);
+		if (frameBufferSize != screenSize) {
+			MmPersistContiguousMemory(framebufferMemory, frameBufferSize, FALSE);
```

Comment:
In case you didn't already see it, #507 seems to be triggered by freeing the saved framebuffer when an XBE is launched by the dashboard. 
I'm wondering if the dashboard expects to have the framebuffer at a specific address and some work needs to be done to relocate mem such that the new framebuffer allocated later in this method can be put at the same addr?

Reply:
I know the dashboard retrieves the address from `AvGetSavedDataAddress`, so that shouldn't be an issue.
I do have a suspicion what the problem could be, though, and will try to confirm it later today.